package/gluon-core/luasrc/usr/bin/gluon-info

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local pretty_hostname = require 'pretty_hostname'
+
+local site = require 'gluon.site'
+local sysconfig = require 'gluon.sysconfig'
+local platform = require 'gluon.platform'
+local util = require "gluon.util"
+local has_vpn, vpn = pcall(require, 'gluon.mesh-vpn')
+
+function _(v)
+  return v
+end
+
+local pubkey
+if has_vpn and vpn.enabled() then
+  local _, active_vpn = vpn.get_active_provider()
+
+  if active_vpn ~= nil then
+    pubkey = active_vpn.public_key()
+  end
+
+  if pubkey == '' then
+    pubkey = nil
```

Comment:
Hmm, it would be great if we could fix the VPN providers never to return an empty string. Maybe they even do that already?

Reply:
This is copypasta from web-admin package, so chances are it's a leftover

Reply:
I'm thinking whether it makes sense to have an exact copy of this and if it's better to move it to some lib instead, that both the UI and CLI source the function from. Perhaps the translate function could be an optional argument then in order to get translations.

Reply:
Hmm, both the fastd and wireguard VPN providers (`gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/provider/fastd.lua`, `gluon-mesh-vpn-wireguard/luasrc/usr/lib/lua/gluon/mesh-vpn/provider/wireguard.lua`) currently return an empty string rather than `nil` when no key exists. I consider this a bug that should be fixed there, so we don't have to handle the empty string in web-admin or in gluon-info.